scripts/helm-nsm-install.sh

```diff
@@ -0,0 +1,170 @@
+    sleep 2
+  done
+  echo "Done"
+
+  # We specifically set admission-webhook variables here as it is a subchart
+  # there might be a way to set these as global and refer to them with .Values.global.org
+  # but that seems more intrusive than this hack. Consider changing to global if the charts
+  # get even more complicated
+  echo "Begin chart installation..."
+  helm install "$CHART" \
+  --atomic --timeout 5m \
+  --set org="$CONTAINER_REPO",tag="$CONTAINER_TAG" \
+  --set forwardingPlane="$FORWARDING_PLANE" \
+  --set insecure="$INSECURE" \
+  --set global.JaegerTracing="true" \
+  --set spire.enabled="$SPIRE_ENABLED",spire.org="$CONTAINER_REPO",spire.tag="$CONTAINER_TAG" \
+  --set admission-webhook.org="$CONTAINER_REPO",admission-webhook.tag="$CONTAINER_TAG" \
+  --set prefix-service.org="$CONTAINER_REPO",prefix-service.tag="$CONTAINER_TAG" \
+  --namespace "$NSM_NAMESPACE" \
+  deployments/helm/"$CHART" &
+  sleep 2
+  while [ -d /proc/$PID ]
+  do
+    printf "."
+    sleep 2
+  done
+  echo "Done"
+}
+
+function usage () {
+  echo "Usage: $0 [flags]"
+  echo "Available flags:"
+  FLAGS="--chart [chart],Chart name. Defaults to value of CHART
+--container_repo [repo URI],Container repository. Defaults to value of CONTAINER_REPO
+--container_tag [tag],Container tag. Defaults to value of CONTAINER_TAG
+--forwarding_plane [forwarding plane],NSM forwarding plane to use. Defaults to value of FORWARDING_PLANE
+--insecure [true|false],Defaults to value of INSECURE
+--nsm_namespace [namespace],Name of the NSM namespace. Defaults to value of NSM_NAMESPACE
+--spire_enabled [true|false],Defaults to value of SPIRE_ENABLED
+-h | --help,Display usage"
+
+echo -e "$FLAGS" | column -t --separator ","
+}
+
+function check_flags () {
+  SHOW_USAGE=0
+  if [ -z ${CHART+x} ]; then
+    echo "CHART is not set, use --chart or set a value for it."
+    SHOW_USAGE=1
+  fi
+  if [ -z ${CONTAINER_REPO+x} ]; then
+    echo "CONTAINER_REPO is not set, use --container_repo or set a value for it."
+    SHOW_USAGE=1
+  fi
+  if [ -z ${CONTAINER_TAG+x} ]; then
+    echo "CONTAINER_TAG is not set, use --container_tag or set a value for it."
+    SHOW_USAGE=1
+  fi
+  if [ -z ${INSECURE+x} ]; then
+    echo "INSECURE is not set, use --insecure or set a value for it."
+    SHOW_USAGE=1
+  fi
+  if [ -z ${FORWARDING_PLANE+x} ]; then
+    echo "FORWARDING_PLANE is not set, use --forwarding_plane or set a value for it."
+    SHOW_USAGE=1
+  fi
+  if [ -z ${NSM_NAMESPACE+x} ]; then
+    echo "NSM_NAMESPACE is not set, use --nsm_namespace or set a value for it."
+    SHOW_USAGE=1
+  fi
+  if [ $SHOW_USAGE -ne 0 ]; then
+    usage
+    exit 1
+  fi
+}
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --chart)
+    CHART="$2"
+    shift
+    shift
+    ;;
+    --container_repo)
+    CONTAINER_REPO="$2"
+    shift
+    shift
+    ;;
+    --container_tag)
+    CONTAINER_TAG="$2"
+    shift
+    shift
+    ;;
+    --insecure)
+    INSECURE="$2"
+    shift
+    shift
```

Comment:
If `INSECURE` env is not set, then installation scripts hangs forever because of the second `shift`. Please add `INSECURE?=false` to Makefiles (same way as for `CONTAINER_REPO` or `CONTAINER_TAG`)

Reply:
Thanks for pointing this out, I'll fix that.